File: solutions/javascript/rna-transcription/rna-transcription.js
/**
 * Determine the RNA complement of a given DNA sequence.
 *
 * @param {string} dna
 * @returns {string}
 */
export const toRna = (dna) => {
  /**
   * @type {Record<string,string}
   */
  const DNA2RNA = {
    G: "C",
    C: "G",
    T: "A",
    A: "U",
  };
  return dna
    .split("")
    .map((nucleotide) => DNA2RNA[nucleotide])
    .join("");
};

File: solutions/python/rna-transcription/rna_transcription.py
"""RNA Transcription"""

DNA2RNA = {"G": "C", "C": "G", "T": "A", "A": "U"}


def to_rna(dna_strand):
    """Given a DNA strand, return its RNA Complement Transcription."""

    return "".join([DNA2RNA[nucleotide] for nucleotide in dna_strand])
